modules/apps/segments/segments-api/src/main/java/com/liferay/segments/manager/SegmentsExperienceManager.java

```diff
@@ -27,6 +28,14 @@ public SegmentsExperienceManager(
 	}
 
 	public long getSegmentsExperienceId(HttpServletRequest httpServletRequest) {
+		long segmentsExperienceId = ParamUtil.getLong(
+			PortalUtil.getOriginalServletRequest(httpServletRequest),
+			"segmentsExperienceId", -1);
+
+		if (segmentsExperienceId != -1) {
+			return segmentsExperienceId;
+		}
+
 		long[] segmentsExperienceIds = GetterUtil.getLongValues(
 			httpServletRequest.getAttribute(
 				SegmentsWebKeys.SEGMENTS_EXPERIENCE_IDS));
```

Comment:
Hi @felipelorenz, checking `segmentsExperienceId` will actually override the logic involving `SegmentsWebKeys.SEGMENTS_EXPERIENCE_IDS` and I don't think we want this.

From what I saw, the issue is happening because the user is not a member of the segment linked to the experience he is trying to see.

A better solution would be checking if user has permission to edit the page (I believe this permission should allow the user to change the experience) in [BaseSegmentsEntryProvider::getSegmentsEntryIds](https://github.com/liferay/liferay-portal/blob/018c46ee4f7af1eb39b9cf0ec93f43f92aabdb3a/modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/provider/BaseSegmentsEntryProvider.java#L119)

Reply:
I examined the code but didn't understand the changes you were suggesting and where they would be placed. As time is crucial now for the customer ticket, I'm leaving the fix up to this team.

Reply:
Ok, I just got some context from @felipelorenz and understood the use case you are covering here. Therefore, this change looks good to me. My only point is that this attribute should only be taken into account when the user is a site admin, otherwise, any user who sets this parameter will see an experience that is not expected for him.
I'll add this condition and forward.
Thanks @joshuacords and @felipelorenz